OmniKit/PumpManager/OmnipodPumpManager.swift

```diff
@@ -1364,32 +1364,64 @@ extension OmnipodPumpManager: PumpManager {
                 state.bolusEngageState = .engaging
             })
 
-            var podStatus: StatusResponse
-
-            do {
-                podStatus = try session.getStatus()
-            } catch let error {
-                completion(.failure(PumpManagerError.communication(error as? LocalizedError)))
-                return
-            }
-
             // If pod suspended, resume basal before bolusing
-            if podStatus.deliveryStatus == .suspended {
+            if case .some(.suspended) = self.state.podState?.suspendState {
                 do {
                     let scheduleOffset = self.state.timeZone.scheduleOffset(forDate: Date())
                     let beep = self.confirmationBeeps
-                    podStatus = try session.resumeBasal(schedule: self.state.basalSchedule, scheduleOffset: scheduleOffset, acknowledgementBeep: beep, completionBeep: beep)
+                    _ = try session.resumeBasal(schedule: self.state.basalSchedule, scheduleOffset: scheduleOffset, acknowledgementBeep: beep, completionBeep: beep)
                 } catch let error {
+                    self.log.error("enactBolus: error resuming suspended pod: %s", String(describing: error))
                     completion(.failure(PumpManagerError.communication(error as? LocalizedError)))
                     return
                 }
             }
 
-            guard !podStatus.deliveryStatus.bolusing else {
-                completion(.failure(PumpManagerError.communication(PodCommsError.unfinalizedBolus)))
+            var doGetStatus = false
```

Comment:
Reading below, it looks like this `doGetStatus` is set when you want to attempt an uncertain bolus reconciliation?  If so, maybe we can give it a more descriptive name, like `attemptUncertainBolusReconciliation`, or make a comment; this method is getting fairly hairy and hard to read.

Reply:
Yes, Loop still needs to the get status before a bolus to attempt an uncertain bolus reconciliation as the self.log.info("enactBolus: doing getStatus with uncertain bolus scheduled certainty") call right before setting doGetStatus documented in lieu of a comment. Nevertheless, I'll rename this variable to `attemptUncertainBolusReconciliation` to be even more explicit about the reason for needing to do the get status for this case.